executor/distsql.go

```diff
@@ -195,45 +177,26 @@ func indexRangesToKVRanges(tid, idxID int64, ranges []*types.IndexRange) ([]kv.K
 	return krs, nil
 }
 
-func extractHandlesFromIndexResult(idxResult distsql.SelectResult) (handles []int64, finish bool, err error) {
-	subResult, e0 := idxResult.Next()
-	if e0 != nil {
-		err = errors.Trace(e0)
-		return
-	}
-	if subResult == nil {
-		finish = true
-		return
-	}
-	handles, err = extractHandlesFromIndexSubResult(subResult)
-	if err != nil {
-		err = errors.Trace(err)
-	}
-	return
-}
-
-func extractHandlesFromIndexSubResult(subResult distsql.PartialResult) ([]int64, error) {
-	defer terror.Call(subResult.Close)
-	var (
-		handles     []int64
-		handleDatum types.Datum
-	)
-	handleType := types.NewFieldType(mysql.TypeLonglong)
-	for {
-		data, err := subResult.Next()
-		if err != nil {
-			return nil, errors.Trace(err)
+func (w *indexWorker) extractTaskHandles(chk *chunk.Chunk, idxResult distsql.SelectResult) (handles []int64, err error) {
+	handles = make([]int64, 0, w.batchSize)
+	for len(handles) < w.batchSize {
+		e0 := idxResult.NextChunk(chk)
+		if e0 != nil {
+			err = errors.Trace(e0)
+			return
 		}
-		if data == nil {
-			break
+		if chk.NumRows() == 0 {
+			return
 		}
-		handleDatum, err = tablecodec.DecodeColumnValue(data[0].GetRaw(), handleType, nil)
-		if err != nil {
-			return nil, errors.Trace(err)
+		for i := 0; i < chk.NumRows(); i++ {
+			handles = append(handles, chk.GetRow(i).GetInt64(0))
```

Comment:
1. After the `for` loop, `len(handles)` may be larger than `w.batchSize` and `w.maxBatchSize`
2. Can we add an API for Chunk, like `func (c *Chunk) GetIntVector(colID, begin, end int) []int64`, to get a slice of integers/strings/decimals ...

Reply:
1. I know, but I think it's acceptable that the batch size is a little larger.
2. `GetInt64` is fast enough, it only takes less than 2 ns.